README.md

```diff
@@ -60,22 +60,40 @@ This is included in all configs shared by this plugin, so can be omitted if
 extending them.
 
 The behaviour of some rules (specifically `no-deprecated-functions`) change
```

Comment:
we should create a section here and link to it from https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-deprecated-functions.md